pageserver/src/tenant/delete.rs

```diff
@@ -148,7 +148,8 @@ async fn ensure_timelines_dir_empty(timelines_path: &Utf8Path) -> Result<(), Del
     // Assert timelines dir is empty.
     if !fs_ext::is_directory_empty(timelines_path).await? {
         // Display first 10 items in directory
-        let list = &fs_ext::list_dir(timelines_path).await.context("list_dir")?[..10];
+        let list = fs_ext::list_dir(timelines_path).await.context("list_dir")?;
+        let list = &list.into_iter().take(10).collect::<Vec<_>>();
```

Comment:
hmm we could avoid the recollect here, use iter().take(10). the original list will live until string is formatted.

Reply:
It's slightly terser to iter() instead of into_iter(), but I generally prefer the simplicity of moving rather than referencing.  Kind of doesn't matter either way in this context though.

Reply:
And avoid allocating.